feat: add WebAuthn registration endpoints to ShopperLogin

hajinsuha1 opened the pull request:
- Added three new WebAuthn registration mutation endpoints to ShopperLoginMutations:
  - startWebauthnRegistration: Generates credential creation options and challenge
  - finishWebauthnRegistration: Verifies and stores authenticator credentials
  - authorizeWebauthnRegistration: Validates user and creates registration token
- Updated cache configuration to handle new WebAuthn mutations with noop cache updates
- Added JSDoc comments describing

<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

hajinsuha1 committed:
feat: add WebAuthn registration endpoints to ShopperLogin

- Added three new WebAuthn registration mutation endpoints to ShopperLoginMutations:
  - startWebauthnRegistration: Generates credential creation options and challenge
  - finishWebauthnRegistration: Verifies and stores authenticator credentials
  - authorizeWebauthnRegistration: Validates user and creates registration token
- Updated cache configuration to handle new WebAuthn mutations with noop cache updates
- Added JSDoc comments describing

cc-prodsec commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/f0d7c7e3-f7cd-4c91-b139-56eff6d2439e?refs=6f994ed8094e13d8647bcdd154bbdccc7d2b290a&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/f0d7c7e3-f7cd-4c91-b139-56eff6d2439e/license?refs=6f994ed8094e13d8647bcdd154bbdccc7d2b290a&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

hajinsuha1 committed:
feat: add WebAuthn authentication endpoints to ShopperLogin

- Added startWebauthnAuthentication endpoint to initiate WebAuthn passkey authentication process
- Added finishWebauthnAuthentication endpoint to complete authentication and return OAuth tokens
- Updated cache update matrix to include new WebAuthn authentication endpoints with noop handlers

hajinsuha1 committed:
add unit tests in mutation.test.ts
